interlok-core/src/main/java/com/adaptris/core/services/MessageUniqueIdService.java

```diff
@@ -0,0 +1,34 @@
+package com.adaptris.core.services;
+
+import com.adaptris.core.AdaptrisMessage;
+import com.adaptris.core.CoreException;
+import com.adaptris.core.ServiceException;
+import com.adaptris.core.ServiceImp;
+import com.adaptris.core.util.MessageHelper;
+
+/**
+ * This service generates a new message unique id and sets it on the message
+ */
+public class MessageUniqueIdService extends ServiceImp {
```

Comment:
For this one, I was probably not clear with my Friday last minute email.
We need something where users can configure which unique Id they want.
So with something like:

```xml
<message-unique-id-service>
  <message-unique-id>NewMessageId</message-unique-id>
</message-unique-id-service>
```

And also we should be able to use expression to get the value from metadata so something like that:
```xml
<message-unique-id-service>
  <message-unique-id>%message{metadataWithMessageId}</message-unique-id>
</message-unique-id-service>
```
Let me know if you need help with that.

And finally, have a look at https://interlok.adaptris.net/interlok-docs/#/pages/developer/developer-annotations to annotate the service. to work nicely in the UI.